### PR TITLE
[Webapp] Bugfix: az webapp vnet-integration add - fix error handling if bad vnet name, support vnet resource ID

### LINKED ISSUE #13863: az webapp vnet-integration add fails with api-version '2020-04-01' is invalid
**Describe the bug**
`az webapp vnet-integration add` fails (at least in west europe) with message
```
Command group 'webapp vnet-integration' is in preview. It may be changed/removed in a future release.
The api-version '2020-04-01' is invalid. The supported versions are '2020-05-01,2020-01-01,2019-11-01,2019-10-01,2019-09-01,2019-08-01,2019-07-01,2019-06-01,2019-05-10,2019-05-01,2019-03-01,2018-11-01,2018-09-01,2018-08-01,2018-07-01,2018-06-01,2018-05-01,2018-02-01,2018-01-01,2017-12-01,2017-08-01,2017-06-01,2017-05-10,2017-05-01,2017-03-01,2016-09-01,2016-07-01,2016-06-01,2016-02-01,2015-11-01,2015-01-01,2014-04-01-preview,2014-04-01,2014-01-01,2013-03-01,2014-02-26,2014-04'.
```

UPDATE: This happens only if `--vnet` param is passed in form of resourceId not name.

**To Reproduce**
Run `az webapp vnet-integration add` for a web app in west europe.

**Expected behavior**
Should work :-).

**Environment summary**
apt-get/2.7.0/Ubuntu 18.04/Bash on Windows

**Additional context**
west europe

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**  
closes #13863, #11454 , #12876

Currently, `az webapp vnet-integration add` command gives a non descriptive error when an invalid Vnet name is passed using --vnet. The error shown is: 'The api-version 2020-04-01 is in valid...' 
Also, --vnet parameter should support using both vnet name and vnet resource ID. Currently, it only supports using vnet name

**Testing Guide**  
- `az webapp vnet-integration add -g RG -n WebappName --vnet=nonExistentVnet --subnet=default` should return error stating that vnet is not found. Currently, it fails with error saying 'The api-version 2020-04-01 is in valid...'
- `az webapp vnet-integration add -g RG -n WebappName --vnet="/subscriptions/pathToVnetResourceId" --subnet=default` should work. Currently, command does not support using Vnet resource ID

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Webapp

**Comment by panchagnula** [comment]
@calvinsID loos good :)
 can you add test cases for adding via  Vnetesource ID & also the case where subnet name is duplicate.
How does the command currently handle same subnet name across different RGs?

**Comment by yungezz** [comment]
hi @qwordy , could you pls look at the PR?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -3223,21 +3223,31 @@ def add_vnet_integration(cmd, name, resource_group_name, vnet, subnet, slot=None
...
-    vnet_resource_group = ''
-    i = 0
-    for z in vnet_id_strings:
-        if z.lower() == "resourcegroups":
-            vnet_resource_group = vnet_id_strings[i + 1]
-        i = i + 1
-
+        if vnet in (v.name, v.id):
+            # parsing the arm uri in order to extract vnet_name and vnet_resource_group
+            vnet_id_strings = v.id.split('/')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:3230]: you can use parse_resource_id common utility, example https://github.com/Azure/azure-cli/blob/8adf97355a605d147afc26f6c1b6f6afb6651c2f/src/azure-cli/azure/cli/command_modules/hdinsight/util.py#L14

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -620,18 +620,18 @@ def load_arguments(self, _):
 
     with self.argument_context('webapp vnet-integration') as c:
         c.argument('name', arg_type=webapp_name_arg_type, id_part=None)
-        c.argument('slot', help="the name of the slot. Default to the productions slot if not specified")
-        c.argument('vnet', help="Vnet name",
+        c.argument('slot', help="The name of the slot. Default to the productions slot if not specified")
+        c.argument('vnet', help="The name or resource ID of the Vnet",
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:624]: [thumb up]

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Thanks!

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 3]
```diff
@@ -620,18 +620,18 @@ def load_arguments(self, _):
...
                    local_context_attribute=LocalContextAttribute(name='vnet_name', actions=[LocalContextAction.GET]))
-        c.argument('subnet', help="Subnet name",
+        c.argument('subnet', help="The name of the subnet",
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:626]: Does it support ID?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 4]
```diff
@@ -3223,21 +3223,26 @@ def add_vnet_integration(cmd, name, resource_group_name, vnet, subnet, slot=None
...
-
+        if vnet in (v.name, v.id):
+            vnet_details = parse_resource_id(v.id)
+            vnet_resource_group = vnet_details['resource_group']
+            vnets.append((v.id, v.name, vnet_resource_group))
+
+    if not vnets:
+        return logger.warning("The virtual network %s was not found in the subscription.", vnet)
+
+    # If more than one vnet, try to use one from same resource group. Otherwise, use first and log the vnet resource id
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:3236]: Add this line to help?

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 3 above)
Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:626]: No, only vnet resource ID is supported

**Comment by panchagnula** [comment]
@calvinsID is this ready for re-review?

**Comment by calvinsID** [comment]
@panchagnula ready!
